aws-ssm-parameter/src/main/java/com/amazonaws/ssm/parameter/BaseHandlerStd.java

```diff
@@ -1,94 +1,161 @@
-    }
+
+	private final SsmClient ssmClient;
+
+	protected BaseHandlerStd() {
+		this(ClientBuilder.getClient());
+	}
+
+	protected BaseHandlerStd(SsmClient ssmClient) {
+		this.ssmClient = requireNonNull(ssmClient);
+	}
+
+	private SsmClient getSsmClient() {
+		return ssmClient;
+	}
+
+	protected static final Set<String> THROTTLING_ERROR_CODES = ImmutableSet.of(
+		"ThrottlingException",
+		"TooManyUpdates");
+
+	@Override
+	public ProgressEvent<ResourceModel, CallbackContext> handleRequest(final AmazonWebServicesClientProxy proxy,
+		final ResourceHandlerRequest<ResourceModel> request,
+		final CallbackContext callbackContext,
+		final Logger logger) {
+		return handleRequest(
+			proxy,
+			request,
+			Optional.ofNullable(callbackContext).orElse(new CallbackContext()),
+			proxy.newProxy(ClientBuilder::getClient),
+			logger);
+	}
+
+	protected abstract ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+		AmazonWebServicesClientProxy proxy,
+		ResourceHandlerRequest<ResourceModel> request,
+		CallbackContext callbackContext,
+		ProxyClient<SsmClient> client,
+		Logger logger);
+
+	protected Constant getBackOffDelay(final ResourceModel model) {
+		if (model.getDataType() != null && model.getDataType() == Constants.AWS_EC2_IMAGE_DATATYPE) {
+			return Constant.of()
+				.timeout(Duration.ofMinutes(5))
+				.delay(Duration.ofSeconds(30))
+				.build();
+		} else {
+			return Constant.of()
+				.timeout(Duration.ofMinutes(5))
+				.delay(Duration.ofSeconds(5))
+				.build();
+		}
+	}
+
+	/**
+	 * If your resource requires some form of stabilization (e.g. service does not provide strong
+	 * consistency), you will need to ensure that your code accounts for any potential issues, so that
+	 * a subsequent read/update requests will not cause any conflicts (e.g.
+	 * NotFoundException/InvalidRequestException) for more information ->
+	 * https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-test-contract.html
+	 *
+	 * @param putParameterRequest  the aws service request to create a resource
+	 * @param putParameterResponse the aws service response to create a resource
+	 * @param proxyClient          the aws service client to make the call
+	 * @param resourceModel        resource model
+	 * @param callbackContext      callback context
+	 * @return boolean state of stabilized or not
+	 */
+	protected Boolean stabilize(
+		final PutParameterRequest putParameterRequest,
+		final PutParameterResponse putParameterResponse,
+		final ProxyClient<SsmClient> proxyClient,
+		final ResourceModel resourceModel,
+		final CallbackContext callbackContext,
+		final Logger logger
+	) {
+		try {
+			logger.log(String.format("Trying to stabilize %s [%s]", ResourceModel.TYPE_NAME, putParameterRequest.name()));
+			GetParametersResponse response = proxyClient.injectCredentialsAndInvokeV2(Translator.getParametersRequest(resourceModel), proxyClient.client()::getParameters);
+
```

Comment:
Just check if we need version while stabilization

Reply:
version is already taken care of in stabilization. 
PUT Response is sent to stabilization method, and GET response version is compared with PUT response version.